### PR TITLE
Adding before-save-hook which runs buildifier.

### LINKED ISSUE #5: FR: provide buildifier save hooks
It'd be great to have formatting on save!

### PR DESCRIPTION
**Pull request by tautomaton**
Changes:
- Adds a customizable boolean bazel-mode-buildifier-before-save.
- Adds bazel-mode--buildifier-before-save-hook, which runs buildifier if bazel-mode-buildifier-before-save is true, and adds this as a before-save-hook to bazel-mode.

Fixes #5

### CODE REVIEW

**Comment by tautomaton** [comment]
Friendly ping.

File: bazel-mode.el [hunk 1]
```diff
@@ -33,6 +33,13 @@
   :link '(url-link
           "https://github.com/bazelbuild/buildtools/tree/master/buildifier"))
 
+(defcustom bazel-mode-buildifier-before-save nil
+  "Specifies whether to run buildifer in 'before-save-hook'."
+  :type 'boolean
+  :group 'bazel-mode
+  :link '(url-link
```
Comment (phst) [inline bazel-mode.el]: This should probably be marked as risky since buildifier is an external tool that could do anything.

File: bazel-mode.el [hunk 2]
```diff
@@ -76,7 +88,8 @@
   (setq-local comment-start-skip "#+")
   (setq-local comment-end "")
   (setq-local comment-use-syntax t)
-  (setq-local font-lock-defaults '(nil)))
+  (setq-local font-lock-defaults '(nil))
+  (add-hook 'before-save-hook 'bazel-mode--buildifier-before-save-hook))
```
Comment (phst) [inline bazel-mode.el]: nit: use `#'` to refer to function symbols so that the byte compiler complains if the function isn't defined.

Comment (phst) [inline bazel-mode.el]: This changes the global hook. You either need to change the local hook or check for the mode in the hook function. Otherwise this will attempt to format every visited file with Buildifier.

File: bazel-mode.el (hunk 1 above)
Comment (tautomaton) [reply bazel-mode.el]: Done. I also changed bazel-mode-buildifier-cmd to bazel-mode-buildifier-command as https://www.gnu.org/software/emacs/manual/html_node/elisp/File-Local-Variables.html indicates this will automatically mark it as risky as well.

File: bazel-mode.el (hunk 2 above)
Comment (tautomaton) [reply bazel-mode.el]: Done.

Comment (tautomaton) [reply bazel-mode.el]: Done.

**Comment by tautomaton** [comment]
Is this code ready to merge?

File: bazel-mode.el [hunk 3]
```diff
@@ -60,6 +68,11 @@
       (delete-file buildifier-input-file)
       (delete-file buildifier-error-file))))
 
+(defun bazel-mode--buildifier-before-save-hook ()
+  "Run buildifer as a 'before-save-hook'."
```
Comment (phst) [inline bazel-mode.el]: Emacs convention is `` `before-save-hook' `` (or with curly quotes if you prefer that).

Also nit: `before-save-hook` is the entire hook, so this should rather say "Function to be added to before-save-hook" or so.

File: bazel-mode.el [hunk 4]
```diff
@@ -26,13 +26,21 @@
   :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
   :group 'languages)
 
-(defcustom bazel-mode-buildifier-cmd "buildifier"
+(defcustom bazel-mode-buildifier-command "buildifier"
```
Comment (phst) [inline bazel-mode.el]: I'd do this rename in a separate commit since it's quite unrelated

File: bazel-mode.el (hunk 3 above)
Comment (tautomaton) [reply bazel-mode.el]: Done.

File: bazel-mode.el (hunk 4 above)
Comment (tautomaton) [reply bazel-mode.el]: Done.

**Comment by tautomaton** [comment]
Laurent, is this PR ready to merge?
